src/opendr/planning/end_to_end_planning/e2e_planning_learner.py

```diff
@@ -85,8 +90,8 @@ def fit(self, env=None, logging_path='', silent=False, verbose=True):
         self.env = Monitor(self.env, filename=self.logdir)
         self.env = DummyVecEnv([lambda: self.env])
         self.agent.set_env(self.env)
-        self.agent.learn(total_timesteps=self.iters, callback=self.callback)
-        return {"last_20_episodes_mean_reward": self.mean_reward}
+        checkpoint_callback = CheckpointCallback(save_freq=1000, save_path=self.logdir, name_prefix='rl_model')
+        self.agent.learn(total_timesteps=self.iters, callback=checkpoint_callback)
 
     def eval(self, env):
         """
```

Comment:
Below on line 107 `# env = Monitor(env, filename=self.logdir)` could be removed if not needed.

Reply:
removed!